fix: range not value not showing on safari

oozorataiyou opened the pull request:
Fixes #14
I've also commented on the issue the fix itself here is a pull request instead, same solution.

oozorataiyou commented:
I am working on a bootstrap version of the UI,  do tell me if you are interested 
I've attached a preview of what it would look like
![Screenshot 2020-09-07 at 16 34 40](https://user-images.githubusercontent.com/3457550/92366706-16b26f80-f128-11ea-92be-524a70f01284.png)

open-dynaMIX commented on webui-page/webui.css:
```diff
@@ -183,9 +183,9 @@ input[type="range"] {
-  width: 0; /* 1 */
-  height: 40px;
-  background: #fff;
+  width: 1; /* 1 */
+  height: 85px;
+  background: #999;
```
```suggestion
  width: 50px; /* 1 */
  height: 85px;
  background: var(--main-fg-color);
```

open-dynaMIX reviewed (changes requested):
Hey @oozorataiyou 

Thanks a lot for your contribution :tada:

I think 1px width is too small. Additionally, I'd lilke to have the `main-fg-color` as background. See my suggestion below.

Unfortunately, this doesn't fully fix #161. On Android and desktop it's possible to just touch/click anywhere in the bar to jump there. It would be awesome to achieve that.

oozorataiyou commented:
Hi, I am unsure what you mean by this:
> On Android and desktop it's possible to just touch/click anywhere in the bar to jump there. It would be awesome to achieve that.

I've tested on the following devices and browsers:
Safari: iOS 13.6.1 and iOS 14.0 (beta 7), macOS Big Sur beta 6
Chrome: iOS 14.0 (beta 7), android, macOS Big Sur beta 6

clicking on the range it will jump to the location, so does dragging the thumb.
May I know the device and/or iOS version that you are using so I can test on the simulator?
Or do you not want it to work?

I will send a pull request of the newer version once I understand the bug you are talking about and try and fix it if possible.
I just thought it was a visual problem with safari so I just set it to 1 thinking you wanted it to look more like a progressbar with the CSS that was written, thus 1px was enough to make the bar show up and not disturbing your UI too much.

open-dynaMIX commented:
The only iOS device I have availabe is an old iPad with iOS 10.3.3. It's possible, that it works on newer versions, idk. It would be good enough for me if it works on modern devices, but then I'd have no way of testing.

open-dynaMIX commented:
> I am working on a bootstrap version of the UI, do tell me if you are interested

Looks nice! What's most important for me about the frontend is, that it doesn't use any external resouces, so that no internet connection is needed. Additionally I'd prefer not having any build step involved for the UI.

In case you didn't know: With the `static_dir` option, you can set the path from where the frontend is served.

oozorataiyou commented:
Hi I would like to explain on why made some of the changes I made.
Firstly on the the main point of not changing it to `50px`,  changing it to `1px` is to ensure that the bar is shown on iOS Safari for it to look the same as much as possible I changed it to `1px`, this is the minimum for the bar to show (this was the bug I tot you were referring to at first), it was 0 at first. 
Next I changed the sliders to use a const height, sorry it bugs me when vars used many times over are not "const".
I added touch events to ensure that touch any where on the sliders and drag would work. (the problem you were probably actually referring to)
I hope these clears up some of the code, ask me anything if you don't understand why I did it.

oozorataiyou commented:
> Additionally I'd prefer not having any build step involved for the UI.

Could you elaborate what **build steps** is sorry.

I've added the min files needed for bootstrap in the version I am working on so there shouldn't be any build process if I didn't misunderstand you. 
But changing to the bootstrap version requires some reworking of the JS especially the playlist part, I will add another branch if possible on my fork to allow you to try it soon if possible, but the rewriting of JS will cause a delay.

open-dynaMIX commented:
Thanks for the update! It seems to work fine now on safari and desktop. Unfortunately on Android the sliders are unusable now. They can't be dragged. And when touched, it's value is read as `NaN` (and also printed to the console) and a 400 is returned from the API.

On another note: I think this change should contain a fix for #14 and nothing more. So please keep the slider height at 85px.

open-dynaMIX commented:
> I've added the min files needed for bootstrap in the version I am working on so there shouldn't be any build process if I didn't misunderstand you.

You've understood me correctly.

> I will add another branch if possible on my fork to allow you to try it soon if possible

:+1:

oozorataiyou commented:
ah, real sorry about that I forgot to test on android. About the 85px really sorry too I copy pasted from my bootstrap version

oozorataiyou commented:
Sorry but I only created a workaround for Android chrome as I can't debug on it

open-dynaMIX commented on webui-page/webui.js:
```diff
@@ -415,6 +426,38 @@ function setPosSlider(position, duration) {
   pos.innerHTML = format_time(slider.value);
 }
 
+function updatePosSlider(position) {
+  if(position === false){
+    return
+  }
+  const slider = document.getElementById("mediaPosition");
+  const pos = document.getElementById("position");
+  slider.value = position;
+  pos.innerHTML = format_time(slider.value);
+}
```
It would be nice to use the `setPostSlider` function above for this. I've made some small changes to it, so you should better rebase first.

open-dynaMIX commented on webui-page/webui.js:
```diff
@@ -444,6 +487,39 @@ function setVolumeSlider(volume, volumeMax) {
   vol.innerHTML = slider.value + "%";
 }
 
+function updateVolSlider(position) {
+  if (position === false) {
+    return
+  }
+  const slider = document.getElementById("mediaVolume");
+  const vol = document.getElementById("volume");
+  slider.value = position;
+  vol.innerHTML = `${position.toFixed(1)}%`;
+}
```
Same applies here. Please use `setVolumeSlider`.

open-dynaMIX commented on webui-page/webui.js:
```diff
@@ -405,6 +405,17 @@ function setMetadata(metadata, playlist, filename) {
 }
 
+function touchRangeOffset(e) {
+  if (!e.layerX){
+    return false
+  }else{
+    e.preventDefault();
+  }
```
```suggestion
  if (!e.layerX){
    return false
  }
  e.preventDefault();
```

open-dynaMIX reviewed (changes requested):
Nice!
It seems to work on android and iOS now, cool!

Just some nitpicks.

oozorataiyou commented:
Hi, 
aside from the javascript I have completed the UI bootstrap too.
I am posting some screenshot here to ask for your advice in the colors of the UI incase if you have any opinions so I can adjust them before creating a new branch :D
Here's a quick link for [Bootstrap Theme colors](https://getbootstrap.com/docs/4.5/getting-started/theming/#theme-colors)

<img width="649" alt="Main Screen" src="https://user-images.githubusercontent.com/3457550/92706700-5034e600-f387-11ea-94d8-913bea725300.png">
<img width="649" alt="Playlist" src="https://user-images.githubusercontent.com/3457550/92704498-92f5be80-f385-11ea-9753-6b35be3be8b8.PNG">
<img width="649" alt="Keyboard shortcuts" src="https://user-images.githubusercontent.com/3457550/92704801-c7697a80-f385-11ea-8b01-8b06f4b8cac0.PNG">

open-dynaMIX commented on webui-page/webui.js:
```diff
@@ -403,6 +403,22 @@ function setMetadata(metadata, playlist, filename) {
+  }
+  e.preventDefault();
+  const slider = e.target;
+  const offset = slider.max * (e.layerX / slider.scrollWidth);
+  return offset;
```
```suggestion
  return slider.max * (e.layerX / slider.scrollWidth);
```

open-dynaMIX reviewed (changes requested):
Cool!

Still some nitpicks and a broken volume slider on iOS (see my suggestion below).

oozorataiyou commented on webui-page/webui.js:
```diff
@@ -411,6 +427,26 @@ function setPosSlider(position, duration) {
+function handleMediaPositionEnd(e) {
+  const offset = touchRangeOffset(e);
+  setPosSlider(offset, document.getElementById("mediaPosition").max);
+  (offset != false) && send("set_position", offset);
```
hi, in `touchRangeOffset(e)` there is a return false, it is to prevent error just incase some lag happens on someone's computer and it doesn't remove the listener in time. 
It was there is my previous update functions too.
I will remove it since you tested and it is now useless.

oozorataiyou commented:
Sorry I didn't rebase the project as I did the updates straight after I woke up, I will be busy today so I can't make any changes anymore for the day.

Honestly, I've never done projects in the team before so I am unsure if you can rebase before you merge the request if you aren't able I will do it tomorrow, so sorry for the small bugs here and there ><
